base/cloudant_base_test.go

```diff
@@ -392,4 +396,382 @@ var _ = Describe(`Cloudant custom base service UT`, func() {
+					"errors": [{
+						"code": "test_value",
+						"message": "test_value: A valid test reason"
+					}]
+				}`,
+			},
+			{
+				description: "Validates no augmentation on successfull response",
+				status:      http.StatusOK,
+				headers:     defaultHeaders,
+				body:        `{"_id": "testdoc", "_rev": "1-abc", "foo": "bar"}`,
+				expect:      `{"_id": "testdoc", "_rev": "1-abc", "foo": "bar"}`,
+			},
+			{
+				description: "Validates no augmentation on HEAD request",
+				status:      http.StatusTeapot,
+				method:      http.MethodHead,
+				headers:     defaultHeaders,
+			},
+			{
+				description: "Validates no augmentation on empty body",
+				status:      http.StatusTeapot,
+				headers:     defaultHeaders,
+				body:        `{}`,
+				expect:      `{}`,
+			},
+			{
+				description: "Validates no augmentation of existing trace",
+				status:      http.StatusTooManyRequests,
+				headers:     defaultHeaders,
+				body: `{
+					"trace": "testanotherreqid",
+					"error": "too_many_requests",
+					"reason": "Buy a bigger plan."
+				}`,
+				expect: `{
+					"trace": "testanotherreqid",
+					"error": "too_many_requests",
+					"reason": "Buy a bigger plan."
+				}`,
+			},
+			{
+				description: "Validates no augmentation of existing 'errors' without trace",
+				status:      http.StatusForbidden,
+				headers:     contentTypeHeader,
+				body: `{
+					"error": "forbidden",
+					"reason": "You must have _reader to access this resource.",
+					"errors": [{
+						"code": "forbidden",
+						"message": "forbidden: You must have _reader to access this resource."
+					}]
+				}`,
+				expect: `{
+					"error": "forbidden",
+					"reason": "You must have _reader to access this resource.",
+					"errors": [{
+						"code": "forbidden",
+						"message": "forbidden: You must have _reader to access this resource."
+					}]
+				}`,
+			},
+			{
+				description: "Validates augmented trace with no augmentation of existing 'errors'",
+				status:      http.StatusForbidden,
+				headers:     defaultHeaders,
+				body: `{
+					"error": "forbidden",
+					"reason": "You must have _reader to access this resource.",
+					"errors": [{
+						"code": "forbidden",
+						"message": "forbidden: You must have _reader to access this resource."
+					}]
+				}`,
+				expect: `{
+					"trace": "testreqid",
+					"error": "forbidden",
+					"reason": "You must have _reader to access this resource.",
+					"errors": [{
+						"code": "forbidden",
+						"message": "forbidden: You must have _reader to access this resource."
+					}]
+				}`,
+			},
+			{
+				description: "Validates no augmentation on none json response",
+				status:      http.StatusBadRequest,
+				headers: map[string]string{
+					"x-couch-request-id": "testreqid",
+					"content-type":       "text/plain",
+				},
+				body:   `foo`,
+				expect: `foo`,
+			},
+			{
+				description: "FIXME! Validates no augmentation on missing content type",
```

Comment:
What needs fixing here?

Reply:
Ah yes, this one is a bit tricky. `httptest` server is always setting `content-type` if it is missing, I tried a couple of thing but without success, so this test is kind of useless. I was planning to to discuss it in review if it worth to dig deeper on how to do this or if we can just remove it.

Reply:
The purpose of the test is to ensure that our error augmentation JSON content-type checking doesn't cause an issue on an invalid response (i.e. we should just ignore it and let pass through and get handled wherever else handles invalid content-types).

Not sure what things you tried, but it sounds like omitting the header completely doesn't work. If an empty string or something invalid like `;` don't work in the test either then we probably have no choice but to drop the test.

I think the code:
```
			ct := resp.Header.Get("Content-Type")
			if strings.HasPrefix(ct, "application/json") 
```
is pretty safe because `Header.Get`'s contract returns `""` for the case where a header doesn't exists so the `HasPrefix` check will fail. I think in some of the other langs we were defending against null/None cases.

Reply:
Yes, omitting header doesn't help, because ["If the Header does not contain a Content-Type line, Write adds a Content-Type set to the result of passing the initial 512 bytes of written data to [DetectContentType]."](https://pkg.go.dev/net/http#ResponseWriter) and sending garbage doesn't help neither, coz `DetectContentType` fails back to "application/octet-stream".

I guess I'll set content-type to empty string then. It wouldn't match other libs where we can check response with content-type, but will at least emulate missing content type, because as you said `Header.Get` returns an empty string when it can't find header.